site/docs/spark-procedures.md

```diff
@@ -240,6 +240,57 @@ Remove any files in the `tablelocation/data` folder which are not known to the t
 CALL catalog_name.system.remove_orphan_files(table => 'db.sample', location => 'tablelocation/data')
 ```
 
+### `rewrite_data_files`
+
+Iceberg tracks each data file in a table. More data files leads to more metadata stored in manifest files, and small data files causes an unnecessary amount of metadata and less efficient queries from file open costs.
+
+Iceberg can compact data files in parallel using Spark with the `rewriteDataFiles` action. This will combine small files into larger files to reduce metadata overhead and runtime file open cost.
+
+#### Usage
+
+| Argument Name | Required? | Type | Description |
+|---------------|-----------|------|-------------|
+| `table`       | ✔️  | string | Name of the table to update |
+| `strategy`    |    | string | Name of the strategy - binpack or sort. Defaults to binpack strategy |
+| `sort_order`  |    | string | Comma separated sort_order_column. Where sort_order_column is a space separated sort order info per column (ColumnName SortDirection NullOrder). <br/> All three members are mandatory to provide for sort_order_column. SortDirection can be ASC or DESC. NullOrder can be FIRST or LAST |
+| `options`     | ️   | map<string, string> | Options to be used for actions|
+| `where`       | ️   | string | predicate as a string used for filtering the files.|
+
+
+See the [`RewriteDataFiles` Javadoc](./javadoc/{{ versions.iceberg }}/org/apache/iceberg/actions/RewriteDataFiles.html#field.summary),
+<br/>  [`BinPackStrategy` Javadoc](./javadoc/{{ versions.iceberg }}/org/apache/iceberg/actions/BinPackStrategy.html#field.summary)
+and <br/> [`SortStrategy` Javadoc](./javadoc/{{ versions.iceberg }}/org/apache/iceberg/actions/SortStrategy.html#field.summary)
+for list of all the supported options for this action.
+
+#### Output
+
+| Output Name | Type | Description |
+| ------------|------|-------------|
+| `rewritten_data_files_count` | int | Number of data which were re-written by this command |
+| `added_data_files_count`     | int | Number of new data files which were written by this command |
+
+#### Examples
+
+Rewrite the data files in table `db.sample` and align data files with table partitioning.
+```sql
+CALL catalog_name.system.rewrite_data_files('db.sample')
+```
+
+Rewrite the data files in table `db.sample` and use sort strategy with id column as sort order column.
+```sql
+CALL catalog_name.system.rewrite_data_files(table => 'db.sample', strategy => 'sort', sort_order => 'id DESC LAST')
```

Comment:
can we make this example with multi-column sort order? like `category ASC NULLS LAST, id DESC`?

btw, I think it's `NULLS LAST` instead of `LAST`?

Reply:
I know. But my procedure argument is already space delimited. so having NULLS LAST will cause parsing problem. So, I called as LAST. Any suggestions ?

Reply:
I also made sort direction and null order mandatory. so that code looks clean at the parsing side. Else as it is string parsing, optional arguments needs lot of checking and validation. 

Multicolumn example, I can update it.

Reply:
hmm... some consistency with SQL language would be ideal, but I don't know if that's too much work to achieve with no clear benefit, any thoughts around this? @RussellSpitzer

Reply:
I can still have it as NULLS_LAST and NULLS_FIRST (instead of space, use underscore) and have document mention these syntax.

Reply:
Done. I have modified and pushed as NULLS_FIRST and NULLS_LAST